src/init.cpp

```diff
@@ -1599,18 +1624,24 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
                 if (!mapBlockIndex.empty() && mapBlockIndex.count(chainparams.GetConsensus().hashGenesisBlock) == 0)
                     return InitError(_("Incorrect or no genesis block found. Wrong datadir for network?"));
 
-                // Check for changed -txindex state
-                if (fTxIndex != gArgs.GetBoolArg("-txindex", DEFAULT_TXINDEX)) {
-                    strLoadError = _("You need to rebuild the database using -reindex to change -txindex");
+                // See if the state of the rewards system is being changed
+                if (fRewardsEnabled != gArgs.GetBoolArg("-rewards", DEFAULT_REWARDS_ENABLED)) {
+                    strLoadError = _("You need to rebuild the database using -reindex to change -rewards");
```

Comment:
This isn't exactly correct, if the user is just trying to turn off rewards right? 
I am thinking about it like so. If a wallet has assetindex enabled. They can use the rewards functionality. If they don't have it enabled, they can't use it. 
I do think there is a reason to have the -rewards, unless I am missing something. 
Let me know.

Reply:
The rewards enablement flag has been removed.